### PR TITLE
Infer features for on demand feature views, support multiple output features

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
Infer features for on demand feature views, support multiple output features 
```

### CODE REVIEW

File: sdk/python/feast/feature_store.py [hunk 1]
```diff
@@ -768,11 +779,17 @@ def get_online_features(
             >>> online_response_dict = online_response.to_dict()
         """
         _feature_refs = self._get_features(features, feature_refs)
-        all_feature_views = self._list_feature_views(
-            allow_cache=True, hide_dummy_entity=False
+        all_feature_views = self._registry.list_feature_views(
+            project=self.project, allow_cache=True
+        )
```
Comment (codyjlin) [inline sdk/python/feast/feature_store.py:784]: I think we decided to move to using FeatureStore versions of `list_feature_views`. In particular, here we wanted to include the info of the dummy entity in this internal version, but hide it from the user-facing functions

**Review by adchia — APPROVED** [review]
Nice! Looks like there is some weird rebasing that happened re the README

File: sdk/python/feast/feature_store.py (hunk 1 above)
Comment (achals) [reply sdk/python/feast/feature_store.py]: thanks, this was useful.

File: sdk/python/feast/type_map.py [hunk 2]
```diff
@@ -72,8 +72,27 @@ def feast_value_type_to_python_type(field_value_proto: ProtoValue) -> Any:
...
+def feast_value_type_to_pandas_type(value_type: ValueType) -> Any:
+    value_type_to_pandas_type: Dict[ValueType, str] = {
+        ValueType.FLOAT: "float",
+        ValueType.INT32: "int",
+        ValueType.INT64: "int",
+        ValueType.STRING: "str",
+        ValueType.DOUBLE: "float",
+        ValueType.BYTES: "bytes",
+        ValueType.BOOL: "bool",
+        ValueType.UNIX_TIMESTAMP: "Timestamp",
```
[the remark this answers is not in the document]
Comment (achals) [reply sdk/python/feast/type_map.py:84]: oops - I think this is a protobuf type. Fixing.

File: sdk/python/feast/feature_store.py [hunk 3]
```diff
@@ -861,32 +878,52 @@ def _augment_response_with_on_demand_transforms(
...
+        all_odfv_feature_names = all_on_demand_feature_views.keys()
+
         if len(all_on_demand_feature_views) == 0:
             return initial_response
         initial_response_df = initial_response.to_df()
+
+        odfv_feature_refs = defaultdict(list)
+        for feature_ref in feature_refs:
+            view_name, feature_name = (
+                feature_ref.split(":")[0],
```
[the remark this answers is not in the document]
Comment (achals) [reply sdk/python/feast/feature_store.py:896]: cool

**Comment by achals** [comment]
> Nice! Looks like there is some weird rebasing that happened re the README

This is a precommit hook that ran on the README. my guess is that it was added after the README changes were committed and it wasn't run on all existing files.

**Review by adchia — APPROVED** [review]
/lgtm

**Pull request merged**